go-controller/pkg/node/egressip.go

```diff
@@ -0,0 +1,210 @@
+// +build linux
+
+package node
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	egressipv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/vishvananda/netlink"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	utilnet "k8s.io/utils/net"
+)
+
+const egressLabel = "egress"
+
+type egressIPLocal struct {
+	gatewayIPv4        string
+	gatewayIPv6        string
+	nodeName           string
+	defaultGatewayIntf string
+}
+
+func (n *OvnNode) watchEgressIP(egressIPLocal *egressIPLocal) error {
+	_, err := n.watchFactory.AddEgressIPHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			eIP := obj.(*egressipv1.EgressIP)
+			if err := egressIPLocal.addEgressIP(eIP); err != nil {
+				klog.Error(err)
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			oldEIP := old.(*egressipv1.EgressIP)
+			newEIP := new.(*egressipv1.EgressIP)
```

Comment:
@alexanderConstantinescu do we want to reflect.DeepEqual() the Spec? eg, wouldn't this cause a lot of flapping if the Status field is updated?